src/main/java/seedu/anilist/logic/parser/ListCommandParser.java

```diff
@@ -0,0 +1,38 @@
+package seedu.anilist.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.anilist.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.anilist.logic.parser.CliSyntax.PREFIX_STATUS;
+import static seedu.anilist.model.Model.PREDICATE_SHOW_ALL_ANIME;
+
+import seedu.anilist.logic.commands.ListCommand;
+import seedu.anilist.logic.parser.exceptions.ParseException;
+import seedu.anilist.model.anime.Status;
+import seedu.anilist.model.anime.StatusEqualsPredicate;
+
+
+public class ListCommandParser implements Parser<ListCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns a ListCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListCommand parse(String args) throws ParseException {
+        if (args.trim().length() == 0) {
+            return new ListCommand(PREDICATE_SHOW_ALL_ANIME);
+        }
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_STATUS);
+        Status statusToMatch;
+
+        if (argMultimap.getValue(PREFIX_STATUS).isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
```

Comment:
Why throw Exception here if no status given? Does this handle the case when user just type in 'list' to view all tab?

Reply:
basically I wanted this to behave in a way such that when you provide extra parameters without specifying status it will fail
`list random stuff` or `list e/episode`, it will fail command.
This is to enforce that status is specified if additional parameters are provided.
`list s/status` or `list random stuff s/status` both works